containerdUtils/containerdUtils.go

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright: (C) 2024 Nestybox Inc.  All rights reserved.
+//
+package containerdUtils
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/BurntSushi/toml"
+)
+
+// Location of containerd config file
+// (see https://github.com/containerd/containerd/blob/main/docs/man/containerd-config.toml.5.md)
+var (
+	configPath = []string{
+		"/etc/containerd/containerd.toml",
+		"/etc/containerd/config.toml",
+		"/usr/local/etc/containerd/config.toml",
+	}
+
+	defaultDataRoot = "/var/lib/containerd"
+)
+
+type containerdConfig struct {
+	Root string `toml:"Root"`
+}
+
+// GetDataRoot returns the containerd data root directory, as read from
+// the containerd config file.
+func GetDataRoot() (string, error) {
+	for _, path := range configPath {
+		dataRoot, err := parseDataRoot(path)
+		if err != nil {
+			if os.IsNotExist(err) {
+				continue
+			}
+			return "", fmt.Errorf("failed to open file %s: %w", path, err)
+		}
+		return dataRoot, nil
+	}
+	return defaultDataRoot, nil
+}
+
+func parseDataRoot(configPath string) (string, error) {
```

Comment:
If we had a linter it would probably complain about this `configPath` function parameter overlapping the global var with the same name. I would rename this parameter to `path`.

Reply:
Good catch, will fix it.

Reply:
Fixed.